docs/developer/build_and_test.html

```diff
@@ -0,0 +1,188 @@
+<!--
+  EVE -  Expressive Vector Engine
+  Copyright 2021 Joel FALCOU
+  Copyright 2021 Jean-Thierry LAPRESTE
+  Copyright 2021 Denis YAROSHEVSKIY
+
+  Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+  SPDX-License-Identifier: MIT
+ -->
+
+ <meta charset="utf-8" lang="en"><style class="fallback">body{visibility:hidden;}</style>
+ **Expressive Vector Engine**
+
+(insert ../crumbs.html here)
+
+# Building EVE: how to do local development.
+
+*NOTE: this documentation was valid at [commit](https://github.com/jfalcou/eve/commit/2fe1048aa31eb3271666f1cca04bf2f5f34eeb6e) and might get outdated.* <br/>
+
+There are two typical ways to build EVE: you can use your local compilers and tools or you can use the docker we use on CI.
+
+## Building for your architecture with your compiler.
+
+*NOTE: this is for linux/mac os. I have not experimented with EVE on Windows.*
+
+First, let's talk about developing locally. Let's say you just want to optimize something for your compiler.
+The tools we use are pretty standard: it's CMAKE and we support ctest.
+The two tricky pieces are: you need c++20 and you need to specify the architecture (march native works for me).
+
+This is how I do cmake setup for my machine:
+
+<script type="preformatted">
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ bash
+  # pwd eve
+  mkdir build && cd build
+  cmake -G Ninja .. '-DCMAKE_CXX_FLAGS=-march=native -g -DEVE_NO_FORCEINLINE' -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+</script>
+
+The only non standard option here is -DEVE_NO_FORCEINLINE. EVE force inlines aggressively
+but it is a massive drag on the compilation speed. So, when working on tests we disable it.
+
+Now we need to actually build and run tests.
+To build and run everything you type:<br/>
+*Adjust -j accordingly to your machine.*
+
+<script type="preformatted">
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ bash
+  # pwd eve/build
+  time ninja -j 16 unit.exe && ctest -j 16
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+</script>
+
+This will run all of the EVE unittests. Which maybe a bit much, depending on what you do.
+
```

Comment:
There we should explain the test naming scheme and target network. I an write that if you want